internal-dns/src/resolver.rs

```diff
@@ -834,8 +833,8 @@ mod test {
         // The DNS server is running, but has no records. Expect a failure.
         let err = client.test_endpoint().await.unwrap_err();
         assert!(
-            err.to_string().contains("no record found"),
-            "Unexpected Error (expected 'no record found'): {err}",
+            err.to_string().contains("error sending request"),
```

Comment:
Why did this bit change? If the server is running, but doesn't have any records, shouldn't the "send part" succeed?

Reply:
I think I can improve this check (insofar as it's useful). The part we were looking for is just farther down the nested chain of errors. I'll do that in a follow on

Reply:
https://github.com/oxidecomputer/omicron/issues/6717